vignettes/what-to-avoid.Rmd

```diff
@@ -0,0 +1,76 @@
+---
+title: "What to avoid"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{what-to-avoid}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup, include = FALSE}
+library(dfeshiny)
+```
+
+The R ecosystem provides a wide range of options to create different elements in dashboard type 
+apps in R-Shiny. Whilst this range of choice can be great for creating anything you want in any 
+style you favour, it also leads to a wide range of pitfalls that can undermine your work.
+
+This document outlines a range of packages and elements to avoid and the reasons why, but the key
+underlying considerations around many of these are:
+
+- Continuity: 
+  - Are functions and packages expected to have long term support from their developers?
+  - Do functions and packages contain complexities that will be difficult to pick up for whoever 
+  succeeds you?
+- Accessibility:
+  - Are the elements created by functions and packages assigned the appropriate HTML tags and 
+  labels?
+  - Do elements created by functions and packages work well with keyboard navigation when in 
+  screen reader mode?
+- Styling
+  - Do functions and packages allow you to create elements that are consistent with the [Government Digital Service style guide](https://www.gov.uk/guidance/style-guide).
```

Comment:
I'd add an extra one around performance / efficiency
- Do they pull in lots of extra dependencies and code you don't need? (thinking of plotly, and how much extra stuff is in there, which is massive overkill if you just want a tooltip)
- How fast are the functions to run? (thinking of plotly again [sorry plotly!] that tends to get a bit slow on bigger datasets / too many plots can bring an app to a halt as they're quite intensive)

Reply:
Added